cli/pcluster/config/validators.py

```diff
@@ -1253,3 +1253,31 @@ def fsx_ignored_parameters_validator(section_key, section_label, pcluster_config
 def _describe_ec2_key_pair(key_pair_name):
     """Return information about the provided ec2 key pair."""
     return boto3.client("ec2").describe_key_pairs(KeyNames=[key_pair_name])
+
+
+def ebs_volume_iops_validator(section_key, section_label, pcluster_config):
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section(section_key, section_label)
+    volume_size = section.get_param_value("volume_size")
+    volume_type = section.get_param_value("volume_type")
+    volume_type_to_iops_ratio = {"io1": 50, "io2": 500}
+    volume_iops = section.get_param_value("volume_iops")
+    min_iops = 100
+    max_iops = 64000
+    if volume_type in volume_type_to_iops_ratio:
+        if volume_iops < min_iops or volume_iops > max_iops:
+            errors.append(
+                "The EBS volume can be provisioned with from {0} IOPS up to {1} IOPS per volume.".format(
+                    min_iops, max_iops
+                )
+            )
+        if volume_iops > volume_size * volume_type_to_iops_ratio[volume_type]:
+            errors.append(
+                "IOPS to volume size ratio of {0} is too high; maximum is {1}.".format(
+                    float(volume_iops) / float(volume_size), volume_type_to_iops_ratio[volume_type]
+                )
+            )
+
+    return errors, warnings
```

Comment:
Even though `volume_size` and `volume_iops` have default values, I would still check both for values of `None` in case we get rid of the default values (as we have discussed doing as part of handling snapshots better).

If these checks aren't performed and the default values are removed, then it's possible that the arithmetic operations used in the validation logic will raise `TypeError`s due to trying to multiply or compare values with an int and `None`.

Thus:
* line 1270 would become `if volume_iops and (volume_iops < min_iops or volume_iops > max_iops):`
* line 1276 would become `if volume_iops and volume_size and volume_iops > volume_size * volume_type_to_iops_ratio[volume_type]:`

Reply:
Yeah! That's right! It is better to check for value of None. Done!